cmds/shutdown/shutdown.go

```diff
@@ -47,17 +65,44 @@ func main() {
 		op = flag.Args()[0]
 	}
 
-	f, ok := opcodes[op]
+	if op == "" {
+		if flags.h && flags.r {
+			return 0, errors.New("Invalid aruments")
+		} else if flags.h {
+			op = "halt"
+		} else {
+			op = "reboot"
+		}
+
+	} else {
+		if flags.h || flags.r {
+			return 0, errors.New("Invalid aruments")
+		}
+	}
+
+	c, ok := opcodes[op]
 	if !ok {
+		return 0, errors.New("Invalid operation")
```

Comment:
Why not tell me what operations are valid?

Reply:
Or at least say "%v is an invalid operation" which can be very helpful for unattended scripts?

Reply:
We print out the usage in main if an error is returned, so that will clarify the valid operations.
That said, it won't hurt to flag up the invalid operation.